src/universal/modules/userDashboard/containers/UserSettings/UserSettingsContainer.js

```diff
@@ -76,7 +88,56 @@ export default class UserSettingsContainer extends Component {
       router.push(nextPage);
     }
     untouch('preferredName');
-  };
+  }
+
+  uploadPicture(pictureFile) {
```

Comment:
I like your thinking here. However, I don't think S3's `PutObject` can callback a URL when the upload is complete. So, unless we send the file data to our server (which I think we want to avoid) we're stuck with client -> S3 -> client -> Parabol -> client.

The way everything works presently, the client can never name the remote S3 filename – they can only use the pre-signed URL the server hands them.